docs/src/main/sphinx/develop/tests.md

```diff
@@ -0,0 +1,440 @@
+includes automatic memory management:
+
+```java
+@TestInstance(PER_METHOD)
+class Test
+{
+
+    @Test
+    public void testSomething()
+    {
+        try (Connection connection = newConnection();) {
+            ...
+        }
+    }
+
+    @Test
+    public void testSomethingElse()
+    {
+        try (Connection connection = newConnection();) {
+            ...
+        }
+    }
+}
+```
+
+## Avoid fake abstractions
+
+Avoid using fake abstraction for tests.
+
+```java
+@DataProvider(name = "data")
+public void test(boolean flag)
+{
+    // WRONG: use separate test methods
+    assertEqual(
+        flag ? ... : ...,
+        flag ? ... : ...);
+}
+```
+
+TBD .. instead do this
+
+```java
+tbd
+```
+
+## Avoid custom parallelization
+
+Do not develop a custom parallel test execution framework:
+
+```java
+@Test(dataProvider = "parallelTests")
+public void testParallel(Runnable runnable)
+{
+   try {
+       parallelTestsSemaphore.acquire();
+   }
+   catch (InterruptedException e) {
+       Thread.currentThread().interrupt();
+       throw new RuntimeException(e);
+   }
+   try {
+       runnable.run();
+   }
+   finally {
+       parallelTestsSemaphore.release();
+   }
+}
+
+@DataProvider(name = "parallelTests", parallel = true)
+public Object[][] parallelTests()
+{
+   return new Object[][] {
+        parallelTest("testCreateTable", this::testCreateTable),
+        parallelTest("testInsert", this::testInsert),
+        parallelTest("testDelete", this::testDelete),
+        parallelTest("testDeleteWithSubquery", this::testDeleteWithSubquery),
+        parallelTest("testUpdate", this::testUpdate),
+        parallelTest("testUpdateWithSubquery", this::testUpdateWithSubquery),
+        parallelTest("testMerge", this::testMerge),
+        parallelTest("testAnalyzeTable", this::testAnalyzeTable),
+        parallelTest("testExplainAnalyze", this::testExplainAnalyze),
+        parallelTest("testRequestTimeouts", this::testRequestTimeouts)
+   };
+}
+```
+
+Leave parallelization to JUnit instead, and implement separate test methods instead.
+
+TBD .. instead do this
+
+```java
+@Test(dataProvider = "testData")
+public void testCreateTable()
+{
+    ...
+}
+
+@Test(dataProvider = "testData")
+public void testInsert()
+{
+    ...
+}
+
+@Test(dataProvider = "testData")
+public void testDelet()
+{
+    ...
+}
+
+...
+```
+
+## Avoid parameterized tests
+
+Do not create a custom parameterized test framework:
+
+```java
+@Test
+public void testTinyint()
+{
+    SqlDataTypeTest.create()
+        .addRoundTrip(...)
+        .addRoundTrip(...)
+        .addRoundTrip(...)
+        .execute(getQueryRunner(), trinoCreateAsSelect("test_tinyint"))
+        .execute(getQueryRunner(), trinoCreateAndInsert("test_tinyint"))
+        .addRoundTrip(...)
+        .execute(getQueryRunner(), clickhouseQuery("tpch.test_tinyint"));
+}
+```
+
+TBD .. instead do this
+
+```java
+tbd
+```
+
+## Original intro from GDOC
```

Comment:
This section is still here for reference only .. I can delete any time .. imho all aspects are covered but wanted to have it here for reviewers for now